pkg/tfbridge/schema.go

```diff
@@ -524,6 +538,11 @@ func MakeTerraformOutput(v interface{},
 		return resource.NewNullProperty()
 	}
 
+	// Marshal sets as their list value.
+	if set, isset := v.(*schema.Set); isset {
+		v = set.List()
+	}
```

Comment:
Totally curious - how is this related?

Reply:
The `MapFieldReader` can produce `*schema.Set` values. `flatmap.Expand` would have instead produced `[]interface{}` values for these properties.